bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+//! Frontend which will be deployed on AssetHub for calling the V2 system pallet
+//! on BridgeHub.
+//!
+//! # Extrinsics
+//!
+//! * [`Call::create_agent`]: Create agent for any kind of sovereign location on Polkadot network.
+//! * [`Call::register_token`]: Register Polkadot native asset as a wrapped ERC20 token on Ethereum.
+#![cfg_attr(not(feature = "std"), no_std)]
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+pub mod weights;
+pub use weights::*;
+
+use frame_support::{pallet_prelude::*, traits::EnsureOrigin};
+use frame_system::pallet_prelude::*;
+use sp_std::prelude::*;
+use xcm::prelude::*;
+use xcm_executor::traits::TransactAsset;
+
+#[cfg(feature = "runtime-benchmarks")]
+use frame_support::traits::OriginTrait;
+
+pub use pallet::*;
+use snowbridge_core::AssetMetadata;
+
+#[derive(Encode, Decode, Debug, PartialEq, Clone, TypeInfo)]
+pub enum ControlCall {
```

Comment:
@acatangiu some context here: The `EthereumSystemFrontend` pallet on AH acts as a proxy for the `EthereumSystem` pallet on BH, to allow users to interact with Snowbridge governance without having to open HRMP calls to BH.

Is this an OK way to specify the call indices for transacting with BH using XCM::Transact?